server/core/src/miscServerFunct.cpp

```diff
@@ -2976,53 +2982,41 @@ irods::error get_script_output_single_line(
-         << "/scripts/" << script_name;
-    } catch (const irods::exception& e) {
-        irods::log(e);
-        return ERROR(-1, "failed to get irods home directory");
-    }
-    for ( std::vector<std::string>::size_type i = 0; i < args.size(); ++i ) {
-        exec << " " << args[i];
-    }
-
-    FILE *fp = popen( exec.str().c_str(), "r" );
-    if ( fp == NULL ) {
-        return ERROR( SYS_FORK_ERROR, "popen() failed" );
-    }
-
-    std::vector<char> buf( 1000 );
-    const char* fgets_ret = fgets( &buf[0], buf.size(), fp );
-    if ( fgets_ret == NULL ) {
-        std::stringstream msg;
-        msg << "fgets() failed. feof["
-            << std::feof( fp )
-            << "] ferror["
-            << std::ferror( fp ) << "]";
-        const int pclose_ret = pclose( fp );
-        msg << " pclose[" << pclose_ret << "]";
-        return ERROR( FILE_READ_ERR, msg.str() );
-    }
-
-    const int pclose_ret = pclose( fp );
-    if ( pclose_ret == -1 ) {
-        return ERROR( SYS_FORK_ERROR,
-                      "pclose() failed." );
+        auto script_path_and_args = args;
+        auto resolved_program = bp::search_path(script_language);
+        if (resolved_program.empty()) {
+            return ERROR(SYS_RESOLUTION_ERROR, fmt::format("Could not resolve {} to an executable", script_language));
+        }
+        auto script_path = fmt::format("{}/scripts/{}", irods::get_irods_home_directory().c_str(), script_name);
+        if (!std::filesystem::exists(script_path)) {
+            return ERROR(SYS_RESOLUTION_ERROR, fmt::format("Script file not found! {}", script_path));
+        }
+        script_path_and_args.insert(
+            script_path_and_args.begin(),
+            fmt::format("{}/scripts/{}", irods::get_irods_home_directory().c_str(), script_name));
+        bp::ipstream program_output;
+        bp::child c(resolved_program, script_path_and_args, bp::std_out > program_output);
+        output.clear();
+        c.wait();
+        const auto ec = c.exit_code();
+        if (!program_output) {
+            return ERROR(ACTION_FAILED_ERR,
+                         fmt::format("stdout was unable to be opened, process exited with code {}", ec));
+        }
+        if (!std::getline(program_output, output)) {
+            return ERROR(ACTION_FAILED_ERR,
+                         fmt::format("Unable to read line from child process, exited with code {}", ec));
+        }
+        return SUCCESS();
     }
-
-    output = &buf[0];
-    // Remove trailing newline
-    const std::string::size_type size = output.size();
-    if ( size > 0 && output[size - 1] == '\n' ) {
-        output.resize( size - 1 );
+    catch (bp::process_error& e) {
+        // In this case, the server is most likely unable to find a suitable
+        // executable to run with a given name
+        irods::experimental::log::server::error("Error starting process, {}", e.what());
+        return ERROR(SYS_RESOLUTION_ERROR, e.what());
```

Comment:
Boost.Process can throw any number of exceptions so we need to use a more general error code here.

For this particular case, `SYS_LIBRARY_ERROR` is likely the best choice. This let's us know that there was a problem using one of the irods-externals libraries.